gc.c

```diff
@@ -2521,12 +2554,13 @@ newobj_init(VALUE klass, VALUE flags, int wb_protected, rb_objspace_t *objspace,
         GC_ASSERT(RVALUE_OLD_P(obj) == FALSE);
         GC_ASSERT(RVALUE_WB_UNPROTECTED(obj) == FALSE);
 
-        if (flags & FL_PROMOTED1) {
-            if (RVALUE_AGE(obj) != 2) rb_bug("newobj: %s of age (%d) != 2.", obj_info(obj), RVALUE_AGE(obj));
+        if (flags & RUBY_FL_PROMOTED) {
+            if (RVALUE_AGE_GET(obj) != 2) rb_bug("newobj: %s of age (%d) != 2.", obj_info(obj), RVALUE_AGE_GET(obj));
```

Comment:
This assertion isn't correct. The original assertion was checking that if `FL_PROMOTED1` was set then the age must be 2 (the object is a candidate).

Reply:
Reading through this part of the code again. I don't think these assertions have much value now that we're pre-tenuring classes in `newobj_init`. I'll delete.